From 1bc9da328a145b8b134f131149995d6d9728df59 Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Sat, 20 Feb 2016 13:23:14 +0000
Subject: [PATCH] editor/edit_dialogs.c: cleanup -Wcast-qual warning

editcmd_dialogs.c:109:44: warning: to be safe all intermediate pointers in cast from 'gchar ** {aka char **}' to 'const char **' must be 'const' qualified [-Wcast-qual]
                 QUICK_RADIO (num_of_types, (const char **) list_of_types,
                                            ^
editcmd_dialogs.c:209:44: warning: to be safe all intermediate pointers in cast from 'gchar ** {aka char **}' to 'const char **' must be 'const' qualified [-Wcast-qual]
                 QUICK_RADIO (num_of_types, (const char **) list_of_types,
                                            ^
editcmd_dialogs.c: In function 'editcmd_dialog_completion_show':
../../lib/widget/widget-common.h:13:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
 #define WIDGET(x) ((Widget *)(x))
                     ^
 editcmd_dialogs.c:349:24: note: in expansion of macro 'WIDGET'
      const Widget *we = WIDGET (edit)
                         ^

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 src/editor/editcmd_dialogs.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/editor/editcmd_dialogs.c b/src/editor/editcmd_dialogs.c
index ed436f8..eb53b1a 100644
--- a/src/editor/editcmd_dialogs.c
+++ b/src/editor/editcmd_dialogs.c
@@ -106,7 +106,7 @@ editcmd_dialog_search_show (WEdit * edit)
                                  INPUT_COMPLETE_NONE),
             QUICK_SEPARATOR (TRUE),
             QUICK_START_COLUMNS,
-                QUICK_RADIO (num_of_types, (const char **) list_of_types,
+                QUICK_RADIO (num_of_types, (const char **)(void *) list_of_types,
                              (int *) &edit_search_options.type, NULL),
             QUICK_NEXT_COLUMN,
                 QUICK_CHECKBOX (N_("Cas&e sensitive"), &edit_search_options.case_sens, NULL),
@@ -206,7 +206,7 @@ editcmd_dialog_replace_show (WEdit * edit, const char *search_default, const cha
                                  "replace", replace_text, NULL, FALSE, FALSE, INPUT_COMPLETE_NONE),
             QUICK_SEPARATOR (TRUE),
             QUICK_START_COLUMNS,
-                QUICK_RADIO (num_of_types, (const char **) list_of_types,
+                QUICK_RADIO (num_of_types, (const char **)(void *) list_of_types,
                              (int *) &edit_search_options.type, NULL),
             QUICK_NEXT_COLUMN,
                 QUICK_CHECKBOX (N_("Cas&e sensitive"), &edit_search_options.case_sens, NULL),
@@ -346,7 +346,7 @@ editcmd_dialog_raw_key_query (const char *heading, const char *query, gboolean c
 char *
 editcmd_dialog_completion_show (const WEdit * edit, int max_len, GString ** compl, int num_compl)
 {
-    const Widget *we = WIDGET (edit);
+    const Widget *we = CWIDGET (edit);
     int start_x, start_y, offset, i;
     char *curr = NULL;
     WDialog *compl_dlg;
